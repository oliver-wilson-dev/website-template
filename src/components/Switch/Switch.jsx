import React from 'react';
import {
  Switch as ReactRouterSwitch,
  Route,
} from 'react-router-dom';

import routes from '../../routes';

const { home, about, contact } = routes;

const Switch = () => (
  <ReactRouterSwitch>
    <Route exact path={home.route}>
      <home.component />
    </Route>
    <Route path={about.route}>
      <about.component />
    </Route>
    <Route path={contact.route}>
      <contact.component />
    </Route>
  </ReactRouterSwitch>
);

export default Switch;
