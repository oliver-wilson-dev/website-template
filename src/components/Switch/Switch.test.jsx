import React from 'react';
import { shallow } from 'enzyme';
import Switch from './Switch';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: ({ children }) => <>{children}</>,
  Route: ({ children }) => <>{children}</>,
  Switch: ({ children }) => <>{children}</>
}));

const HomePage = () => null;
const AboutPage = () => null;
const ContactPage = () => null;

jest.mock('../../routes', () => ({
  home: {
    component: HomePage,
    route: 'test-home-route'
  },
  about: {
    component: AboutPage,
    route: 'test-about-route'
  },
  contact: {
    component: ContactPage,
    route: 'test-contact-route'
  }
}));

const render = () => shallow(<Switch />);

describe('Switch', () => {
  it('renders correctly', () => {
    expect(render()).toMatchSnapshot();
  });
});
