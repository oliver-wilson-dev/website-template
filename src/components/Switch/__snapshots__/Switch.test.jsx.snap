// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<Switch>
  <Route
    exact={true}
    path="test-home-route"
  >
    <HomePage />
  </Route>
  <Route
    path="test-about-route"
  >
    <AboutPage />
  </Route>
  <Route
    path="test-contact-route"
  >
    <ContactPage />
  </Route>
</Switch>
`;
